daffodil-lib/src/main/scala/org/apache/daffodil/lib/api/Diagnostic.scala

```diff
@@ -263,7 +263,12 @@ trait LocationInSchemaFile {
   ): String = {
     // works for both windows and unix
     val splitter = "/"
-    val tokens = uriString.split(splitter)
+    val classpathTrimmedUriString =
+      uriString.replace(
+        Misc.classPath.find(cp => uriString.startsWith(cp.toString)).getOrElse("").toString,
```

Comment:
Same idiom. I found confusing above.

Suggest create a private helper method and call it in both places. That way you can explain it once. 

Name it "stripClasspathPrefix" or something like that perhaps.